invenio_records_resources/services/records/components.py

```diff
@@ -72,3 +51,40 @@ def create(self, identity, data=None, record=None, **kwargs):
     def update(self, identity, data=None, record=None, **kwargs):
         """Inject parsed metadata to the record."""
         record.metadata = data.get('metadata', {})
+
+
+class FilesOptionsComponent(ServiceComponent):
```

Comment:
Major: Let's discuss this during a call today. I think this should be in DraftFilesComponent in Invenio-Drafts-Resources.

Reply:
I saw this in the original ticket and it was mentioned during the call yesterday, but since all of the file logic/responsibility was in here, I went for implementing it here to be reused in drafts-resources. But if we do move *all* the file logic out of records-resources and into drafts-resources then that should work. Otherwise, records-resources still needs to implement files options i.e. doesn't provide everything needed to create records with files.